Source/AliveLibAO/Abe.cpp

```diff
@@ -2039,10 +2040,107 @@ void Abe::FallOnBombs_4231B0()
     }
 }
 
+__int16 Abe::CrouchOrPushWall(const FP gridSize)
+{
+    if (!WallHit_401930(field_BC_sprite_scale * FP_FromInteger(50), gridSize))
```

Comment:
seems like it would make more sense as:

if (WallHit(50))
{
  if (WallHit(20))
  {
    crouch
    return 0;
   }
}
return 1;
?